File: server/InnAiServer/InnAiServer/Dtos/InnAiDataItemDto.cs
namespace InnAiServer.Dtos;

public record InnAiDataDto(int Count, InnAiDataItemDto[] Items);
public record InnAiDataItemDto(DateTime Timestamp, InnLevelDto[] InnLevels, string ImageId);

public record InnLevelDto(int? Level, string Station);

File: innai/InnAiClient/Dtos/InnAiDataItemDto.py
from InnAiClient.Dtos.InnLevelDto import InnLevelDto


class InnAiDataItemDto:
    Timestamp: str
    InnLevels: list[InnLevelDto]
    ImageId: str

    @staticmethod
    def FromJson(json):
        ret = InnAiDataItemDto()
        ret.Timestamp = json["timestamp"]
        ret.InnLevels = [InnLevelDto.FromJson(x) for x in json["innLevels"]]
        ret.ImageId = json["imageId"]
        return ret

